EXT-1022: Border around box when the contrast against the background is low

johannes-lindgren opened the pull request:
When the contrast between the swatch (the colored buttons) and the white background color is low, the swatch will get a border. When the swatch is selected, the color of the checkmark will be ink (black) instead of white.

<img width="310" alt="image" src="https://user-images.githubusercontent.com/14206504/223707817-9d0e23ad-dc18-4fa2-8fa3-f6919157c49f.png">


## Why

When the color is bright, the swatch/button is barely visible, and neither is the checkmark.

![image](https://user-images.githubusercontent.com/14206504/223707742-1f00d31b-5ee9-4c9d-a800-4ec9e095bd41.png)

eunjae-lee reviewed (approved):
This seems quite difficult to maintain, but I guess we won't need to touch those color-related utils in the future. (Is there any library for this in existence?)

johannes-lindgren commented:
The code for calculating the contrast ratio is based on the Web Content Accessibility Guidelines (WCAG) 2.1 9https://www.w3.org/TR/WCAG/). 

I've made the following changes:

- I've created a separate pull request to add Jest to this project: https://github.com/storyblok/field-type-examples/pull/17
- I've added tests for the functions in this pull request so that we can validate that they adhere to the calculations in WCAG 2.1. 
- I've added JSDoc comments, referring to WCAG 2.1, so that it's clear where the calculations originate from.
